modules/storages/app/services/projects/copy/file_links_dependent_service.rb

```diff
@@ -39,31 +39,88 @@ def source_count
     protected
 
     def copy_dependency(*)
-      # If no work packages were copied, we cannot copy their attachments
+      # If no work packages were copied, we cannot copy their file_links
       return unless state.work_package_id_lookup
 
-      state.work_package_id_lookup.each do |old_wp_id, new_wp_id|
-        create_work_package_file_links(old_wp_id, new_wp_id)
-      end
-    end
+      source_wp_ids = state.work_package_id_lookup.keys
+      Storages::FileLink
+        .where(container_id: source_wp_ids, container_type: "WorkPackage")
+        .group_by(&:storage_id)
+        .each do |(storage_id, source_file_links)|
+        if source_file_links.any?
```

Comment:
Maybe use a filtering above, to exclude these cases entirely from `.each` block. That would reduce the nesting and noise.